pkg/apis/core/validation/shoot.go

```diff
@@ -1029,6 +1035,26 @@ func validateKubeletConfigEvictionSoftGracePeriod(eviction *core.KubeletConfigEv
 	return allErrs
 }
 
+// Reserved PIDs have to be non-negative
+const ReservedPIDsMinimum int64 = 0
+
+func validateKubeletConfigReserved(reserved *core.KubeletConfigReserved, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if reserved.CPU != nil {
+		allErrs = append(allErrs, validateResourceQuantityValue("cpu", *reserved.CPU, fldPath.Child("cpu"))...)
+	}
+	if reserved.Memory != nil {
+		allErrs = append(allErrs, validateResourceQuantityValue("memory", *reserved.Memory, fldPath.Child("memory"))...)
+	}
+	if reserved.EphemeralStorage != nil {
+		allErrs = append(allErrs, validateResourceQuantityValue("ephemeralStorage", *reserved.EphemeralStorage, fldPath.Child("ephemeralStorage"))...)
+	}
+	if *reserved.PID < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("pid"), *reserved.PID, fmt.Sprintf("PID must be atleast %d", ReservedPIDsMinimum)))
```

Comment:
See comment in `pkg/apis/core/types_shoot.go`.
If `pid` is also a `resource.Quantity` you can use `validateResourceQuantityValue` here as well.